src/redemptions/redemption.service.ts

```diff
@@ -138,6 +150,59 @@ export class RedemptionService {
     return null;
   }
 
+  async repeatedFaceCheck(
+    response: JumioTransactionRetrieveResponse,
+  ): Promise<{ status: KycStatus; failureMessage: string } | null> {
+    // can only accept if no other errors occurred AND repeated face is from previous transaction of SAME user
+    let potentialFailures: string[] = [];
+    potentialFailures = potentialFailures.concat(
+      response.capabilities.dataChecks.map((i) => i.decision.type),
+      response.capabilities.extraction.map((i) => i.decision.type),
+      response.capabilities.liveness.map((i) => i.decision.type),
+      response.capabilities.similarity.map((i) => i.decision.type),
+      response.capabilities.usability.map((i) => i.decision.type),
+    );
+    // if any other check failed, we can't pass
+    for (const potentialFailure of potentialFailures) {
+      if (potentialFailure !== 'PASSED') {
+        return null;
+      }
+    }
+    // for imageChecks, only duplicate face should pass
+    const failedImageChecks = response.capabilities.imageChecks.filter(
+      (i) => i.decision.type !== 'PASSED',
+    );
+    let duplicateFaceWorkflows: string[] = [];
+    for (const imageCheck of failedImageChecks) {
+      if (imageCheck.decision.details.label !== 'REPEATED_FACE') {
+        return null;
+      }
+      duplicateFaceWorkflows = duplicateFaceWorkflows.concat(
+        imageCheck.data.faceSearchFindings.findings,
+      );
+    }
+    // lookup workflows returned as matching
+    const foundWorkflows = await this.jumioTransactionService.findByWorkflowIds(
+      duplicateFaceWorkflows,
+    );
+
+    // If any are found and don't match current user, ban the current user for scamming
+    for (const foundWorkflow of foundWorkflows) {
+      const currentUserId = Number(response.workflow.customerInternalReference);
+      if (foundWorkflow.user_id !== currentUserId) {
+        return {
+          status: KycStatus.FAILED,
+          failureMessage: `User with multiple accounts detected (previous user id=${foundWorkflow.user_id} , current user id=${currentUserId}), this transaction matches face in transaction_id: ${foundWorkflow.workflow_execution_id}`,
+        };
+      }
+    }
+    return {
+      status: KycStatus.SUBMITTED,
```

Comment:
I woud try to avoid KycStatus at all here - have this be more encapsulated. Input a jumio transaction, and output is repeated face data from that tx.

Then the outer function pulls this data and uses it accordingly.

Reply:
I like this, I will need to split the function into two, because I want to check for intentional pass (two transactions same user) and hard fail (scammer case).